ProjectEureka/project-eureka-flutter
# PE-18 | Google sign in

**What I did:**

- Created a button that allowed new users to sign in with a google account
- Account will be remembered if app is closed
- Sign in processes currently works for Android devices.

**Prerequisites:** 

_Obtain an SHA-1 key and add it to Firebase_

1. Login in to firebase with credentials for PE check drive for more details.
In order to add your SHA-1 key go here:
![image](https://user-images.githubusercontent.com/38572314/105779080-787e4500-5f22-11eb-80af-536170c00d0d.png)
Click on the gear and it should take you to Project settings general.

2. Scroll down to 
![image](https://user-images.githubusercontent.com/38572314/105779187-ac596a80-5f22-11eb-9248-72e41bc8431e.png)
 Click Add fingerprint.

3. Go to your terminal and type. Make sure you're in the right **directory**

- Mac: 

```
keytool -list -v -alias androiddebugkey -keystore ~/.android/debug.keystore
```

- Windows: 
- **Make sure you cd to where debug.keystone is**

```
keytool -list -v -alias androiddebugkey -keystore debug.keystore
```
Default Password: android

4. Once you obtain the key within your terminal add it to firebase and you should be able to login in with a google account. [For more information about getting the SHA-1 ](https://developers.google.com/android/guides/client-auth).

**How to Test:**

1. Make sure emulator or test device is up and running:
![image](https://user-images.githubusercontent.com/38572314/101309213-ae475d00-3800-11eb-959d-977ab77b6d26.png)

2. Run flutter code using command: flutter run or click
![image](https://user-images.githubusercontent.com/38572314/101309131-7cce9180-3800-11eb-8af0-a9d9a0190a5a.png)

3. Click "sign in with Google"

4. Sign-in with a google account. It can be your personal account or a fake account you've created. _Firebase will not show you password._

5. If success: your UID will display in console and you will be taken to the homepage:
![image](https://user-images.githubusercontent.com/38572314/101309717-e1d6b700-3801-11eb-8999-64f945bea7b9.png)
![image](https://user-images.githubusercontent.com/38572314/101309770-02067600-3802-11eb-942a-0a8295ef5e0e.png)
